### PR TITLE
Added a hlint file

### PR DESCRIPTION
**Pull request by aleator**
Added a hlint file for finding places where flow can be used.

It is currently biased to use left-to-right operators

### CODE REVIEW

File: hlint-flow.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+- hint: 
+    lhs: 'f . g'
+    note: 'Use `<.`'
+    rhs: 'f <. g'
+
+
+- hint: 
+    lhs: 'f <| x'
+    note: 'Use `|>` for natural reading direction' 
+    rhs: 'x |> f'
```
Comment (tfausak) [inline hlint-flow.yaml]: I typically use `<|` for passing a big `do` block to a function. Does this hint fire for code like this?

``` hs
liftIO <| do
  now <- getCurrentTime
  print now
```

Comment (aleator) [reply hlint-flow.yaml]: This has been bothering me also as I don't know how to do it. Perhaps this hint should be disabled?

**Comment by aleator** [comment]
You can run this by first installing hlint (stack install works well) and then doing `hlint -h hlint-flow.yaml <file>`.  Would you like me to add few variations to the readme file?

I was impressed about the relude hlint file which was actually super helpful in learning relude and I think flow could be a similar case.

**Comment by tfausak** [comment]
Something in the README would be nice. That's what Relude does. 

We could probably avoid all the directional hints. Maybe keep them in the file but comment them out?

**Comment by aleator** [comment]
I removed the directional hints for `<|` and `< !`. I think the `<.` can still have them. Atleast, nothing where it would conflict with convention immediatedly popped to my mind.

**Pull request merged**